onnxruntime/test/providers/nnapi/nnapi_basic_test.cc

```diff
@@ -403,10 +431,24 @@ TEST(NnapiExecutionProviderTest, TestQDQConcat) {
                           {1, 6, 2},
                       } /* input_shapes */,
                       2 /* axis */),
-                  "nnapi_qdq_test_graph_concat", {
-                                                     true /* verify_entire_graph_use_ep */
-                                                 });
+                  "nnapi_qdq_test_graph_concat",
+                  {ExpectedEPNodeAssignment::All});
+}
+
+#if defined(__ANDROID__)
+TEST(NnapiExecutionProviderTest, TestQDQConcat_UnsupportedInputScalesAndZp) {
+  // This is to verify all the inputs have the same scale and zp as input 0 for API 28-
+  // Currently, this test can only be run locally with a android emulator with API < 29
+  // TODO: consider to configure this and enable it to run in Android CI.
+  const auto* nnapi = NnApiImplementation();
```

Comment:
A note on how to run the emulator this way would be good. Or a pointer to doco on doing so is fine as well.